app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java

```diff
@@ -0,0 +1,130 @@
+package uy.com.bix.app.smsproject.controllers;
+
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+
+import android.preference.PreferenceManager;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import net.danlew.android.joda.JodaTimeAndroid;
+
+import org.joda.time.DateTime;
+
+import uy.com.bix.app.smsproject.classes.AlertReceiver;
+
+public class AlarmController extends AppCompatActivity {
+
+	private static AlarmController instance = null;
+	int expirationYear, expirationMonth, expirationDay, expirationHour, expirationMinute, maxMessages;
+	long dateOfNextFiring;
+	String phoneNumber, textMessage;
+	boolean notifyWhenSending, isLastDay, isActive;
+
+	private AlarmController() {}
+
+	public static AlarmController getInstance() {
+		if (instance == null) {
+			instance = new AlarmController();
+		}
+		return instance;
+	}
+
+	public void prepareAndSendMessage (Context context) {
```

Comment:
what do you mean with "prepare"?
how do you prepare a message?

Here could be a nice javadoc :)